src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversions.cs

```diff
@@ -87,6 +88,234 @@ protected override Conversion GetInterpolatedStringConversion(BoundUnconvertedIn
+            //  * For every _regular_balanced_text_ component of `S` (`Si`) with an _interpolation_format_ component and no _constant_expression_ (alignment) component, overload resolution
+            //    with an identifier of `TryFormatInterpolationHole` and parameter types of `Si` and `string` with name `format` (in that order) succeeds, and contains a single instance
+            //    method that returns a `bool` or `void`.
+            //  * For every _regular_balanced_text_ component of `S` (`Si`) with a _constant_expression_ (alignment) component and no _interpolation_format_ component, overload resolution
+            //    with an identifier of `TryFormatInterpolationHole and parameter types of `Si` and `int` with name `alignment` (in that order) succeeds, and contains a single instance
+            //    method that returns a `bool` or `void`.
+            //  * For every _regular_balanced_text_ component of `S` (`Si`) with an _interpolation_format_ component and a _constant_expression_ (alignment) component, overload resolution
+            //    with an identifier of `TryFormatInterpolationHole` and parameter types of `Si`, `int` with name `alignment`, and `string` with name `format` (in that order) succeeds, and
+            //    contains a single instance method that returns a `bool` or `void`.
+            // Additionally, all resolved method calls must return the same type. `TryFormat` calls that mix `bool` and `void` are not permitted.
+
+            // If the type is applicable, we'll add these to the given info at the end.
+            var useSiteInfo = _binder.GetNewCompoundUseSiteInfo(diagnostics);
+
+            // All builder types have to at least have a `TryFormatBaseString` that takes a single string argument and returns either bool or void.
+            if (!tryGetCandidateMethods("TryFormatBaseString", ref useSiteInfo, out ArrayBuilder<MethodSymbol>? baseStringCandidates))
+            {
+                // PROTOTYPE(interp-strings): We'll want to have a specific error for when we're converting to a non-string type
+                builderArguments = default;
+                diagnostics.AddDiagnostics(source.Syntax, useSiteInfo);
+                return false;
+            }
+
+            var typeArguments = ArrayBuilder<TypeWithAnnotations>.GetInstance();
+            var implicitBuilderReceiver = new BoundImplicitReceiver(source.Syntax, builderType);
+            var analyzedArguments = AnalyzedArguments.GetInstance();
+            var overloadResolutionResult = OverloadResolutionResult<MethodSymbol>.GetInstance();
+            var stringType = _binder.GetSpecialType(SpecialType.System_String, diagnostics, source.Syntax);
+            analyzedArguments.Arguments.Add(new BoundLiteral(CSharpSyntaxTree.Dummy.GetRoot(), constantValueOpt: null, stringType));
```

Comment:
>new BoundLiteral(CSharpSyntaxTree.Dummy.GetRoot(), constantValueOpt: null, stringType) [](start = 44, length = 86)

It doesn't feel like a string null literal is an appropriate argument to use. The spec says a string value, not a null literal.

Reply:
Don't we already have the actual value we are planning to pass to the method?

---
In reply to: [610718396](https://github.com/dotnet/roslyn/pull/52422#discussion_r610718396) [](ancestors = 610718396)

Reply:
> Don't we already have the actual value we are planning to pass to the method?

If there is no content in the string, then no.

---
In reply to: [610719208](https://github.com/dotnet/roslyn/pull/52422#discussion_r610719208) [](ancestors = 610719208,610718396)

Reply:
>> Don't we already have the actual value we are planning to pass to the method?

> If there is no content in the string, then no.

Then why do we care about this method at all? Even if we do, I do not believe using a null literal is the right thing to do.

---
In reply to: [612033734](https://github.com/dotnet/roslyn/pull/52422#discussion_r612033734) [](ancestors = 612033734,610719208,610718396)

Reply:
> Then why do we care about this method at all?

The spec mandates that this method exists at least. Generally, I'm trying to avoid `$""` being convertible to something that really wasn't intended to be a builder and happens to have a method named `Create`, takes 2 int parameters, and returns the current type. That is likely to be a somewhat common thing. We could also word the spec a bit differently to say that member lookup with `AppendLiteral` and `AppendFormat` must have method members?

---
In reply to: [612766918](https://github.com/dotnet/roslyn/pull/52422#discussion_r612766918) [](ancestors = 612766918,612033734,610719208,610718396)